service/test/_registry_test.js

```diff
@@ -796,9 +817,22 @@ describe('Registry', function() {
       <stringified array supplied by the argument devices annotated with importMode property and deviceId property replaced by id>
       ```
       ]*/
+      /* Tests_SRS_NODE_IOTHUB_REGISTRY_06_029: [** A device information with an authentication object that contains a `type` property is considered normalized.] */
+      /* Tests_SRS_NODE_IOTHUB_REGISTRY_06_031: [A device information with an authentication object that doesn't contain the x509Thumbprint property will be normalized with a `type` property with value "sas".] */
+      /* Tests_SRS_NODE_IOTHUB_REGISTRY_06_030: [A device information with an authentication object that contains the x509Thumbprint property will be normalized with a `type` property with value "selfSigned".] */
       it('constructs a valid HTTP request', function(testCallback) {
-        var addedDevices = [{deviceId: 'deviceone'},{deviceId: 'devicetwo'}];
-        var sentBody = [{id:'deviceone',importMode:'create'},{id:'devicetwo',importMode:'create'}];
+        var addedDevices = [
```

Comment:
It is not at though we are comparing individual elements of 1 array to elements of the other.  The entire ARRAY of sent body is compare to what is created from addedDevices.  To put these together makes to me incorrect.

I actually have no idea what you think would be a better layout of some combined structure.  Please use this particular test to provide me with what you think is better.  When you do, I will be happy to change the others.

---
In reply to: [128658351](https://github.com/Azure/azure-iot-sdk-node/pull/61#discussion_r128658351) [](ancestors = 128658351)